### PR TITLE
Add Server Side Telemetry to MSAL

### PR DESCRIPTION
**Pull request by shahzaibj**
Related Common PR: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/671

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 1]
```diff
@@ -1390,6 +1403,7 @@ public void run() {
                             callback
                     );
 
+                    silentTokenCommand.setPublicApiId(PublicApiId.LOCAL_ACQUIRE_TOKEN_SILENT);
```
Comment (rpdome) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: As discussed, 
if you want to track what API is calling, you could rename this function to internal and take an API constant as another parameter, then pass that parameter to setPublicApiId. Should be a pretty straighforward and simple change.

You could do it in this PR or a separate one. That one i'll let you make a call.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 2]
```diff
@@ -1312,6 +1323,7 @@ public void run() {
                             localAuthenticationCallback
                     );
 
+                    command.setPublicApiId(PublicApiId.LOCAL_ACQUIRE_TOKEN_INTERACTIVE);
```
Comment (rpdome) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Same as the comment below.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 3]
```diff
@@ -1064,6 +1067,14 @@ private void initializeTokenSharingLibrary() {
         }
     }
 
+    private void setupServerSideTelemetry(Context context) {
+        EstsTelemetry.initializeEstsTelemetryCache(context);
+        com.microsoft.identity.common.internal.logging.Logger.verbose(
+                TAG,
+                "Server side telemetry cache has been initialized properly."
```
[the remark this answers is not in the document]
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: We aren't really doing much here other than calling constructors and creating objects. The context parameters is non-null, however someone can still pass null to that and in that case the cache won't be initialized. I can add an explicit null check inside the method and also move the log into the method and either log a successful message is cache is created or failure if context is null.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 1 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Yeah, sounds good. I think I'll do that in a separate PR after this gets merged in.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 2 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Replied to the comment above.

File: msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java [hunk 4]
```diff
@@ -334,6 +333,7 @@ public void onTaskCompleted(Boolean result) {
                     }
             );
 
+            command.setPublicApiId(PublicApiId.SIGN_OUT);
```
[the remark this answers is not in the document]
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java]: Addressed

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: I added the log as stated above.

File: msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java [hunk 5]
```diff
@@ -517,10 +518,10 @@ public void acquireTokenSilentAsync(@NonNull final String[] scopes,
     @WorkerThread
     public IAuthenticationResult acquireTokenSilent(@NonNull final String[] scopes,
                                                     @NonNull final String authority) throws MsalException, InterruptedException {
-
         final IAccount persistedAccount = getPersistedCurrentAccount();
         if (persistedAccount == null) {
-            throw new MsalClientException(MsalClientException.NO_CURRENT_ACCOUNT);
+            MsalClientException exception = new MsalClientException(MsalClientException.NO_CURRENT_ACCOUNT);
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java]: why a new variable ?

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java]: I don't remember doing it. I might have accidentally clicked on the `introduce local variable` suggestion within Android Studio and that might have done it. I'll remove it.

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java]: @kreedula Addressed

**Pull request merged**